client/app/bundles/course/survey/__test__/index.test.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Router, createMemoryHistory } from 'react-router';
+import { mount } from 'enzyme';
+import MockAdapter from 'axios-mock-adapter';
+import ReactTestUtils from 'react-addons-test-utils';
+import ProviderWrapper from 'lib/components/ProviderWrapper';
+import CourseAPI from 'api/course';
+import storeCreator from '../store';
+import routes from '../routes';
+
+// Mock axios
+const client = CourseAPI.survey.surveys.getClient();
+const mock = new MockAdapter(client);
+
+beforeEach(() => {
+  mock.reset();
+});
+
+describe('Surveys', () => {
+  it('renders the index page and survey form', async () => {
+    mock.onGet(`/courses/${courseId}/surveys`)
+      .reply(200, {
+        surveys: [],
+        canCreate: true,
+      });
+    // Set the route to index page
+    const history = createMemoryHistory(`/courses/${courseId}/surveys`);
+
+    const spyIndex = jest.spyOn(CourseAPI.survey.surveys, 'index');
+    const store = storeCreator({ surveys: {} });
+
+    const indexPage = mount(
+      <ProviderWrapper store={store}>
+        <Router routes={routes} history={history} />
+      </ProviderWrapper>
+    );
+
+    // Wait for api call
+    await sleep(1);
```

Comment:
Just a random question, is there another way to wait for the API call, through a promise or something?